OpenSim/Simulation/StatesTrajectory.cpp

```diff
@@ -229,3 +283,15 @@ StatesTrajectory StatesTrajectory::createFromStatesStorage(
         const std::string& filepath) {
     return createFromStatesStorage(model, Storage(filepath));
 }
+
+StatesTrajectory::IncompatibleModel::IncompatibleModel(
+        const std::string& file, size_t line,
+        const std::string& func, const Model& model)
+        : OpenSim::Exception(file, line, func) {
+    std::ostringstream msg;
+    auto modelName = model.getName().empty() ? "<empty-name>" :
+                     model.getName();
+    msg << "The provided model '" << modelName << "' is not "
+            "compatible with the StatesTrajectory.";
+    addMessage(msg.str());
+}
```

Comment:
Not for this PR but in future, it will be good to create a hierarchy under `Exception` to represent various kinds of *invalid model* failures.